packages/@aws-cdk/custom-resources/lib/provider-framework/waiter-state-machine.ts

```diff
@@ -49,14 +49,8 @@ export class WaiterStateMachine extends Construct {
     const role = new Role(this, 'Role', {
       assumedBy: new ServicePrincipal('states.amazonaws.com'),
     });
-    role.addToPolicy(new PolicyStatement({
-      actions: ['lambda:InvokeFunction'],
-      resources: [props.isCompleteHandler.functionArn],
-    }));
-    role.addToPolicy(new PolicyStatement({
-      actions: ['lambda:InvokeFunction'],
-      resources: [props.timeoutHandler.functionArn],
-    }));
+    props.isCompleteHandler.grantInvoke(role);
+    props.timeoutHandler.grantInvoke(role);
```

Comment:
Is there a reason why we're okay with `grantInvoke()` here? Or is this legacy and we should be doing the whole

```ts
new PolicyStatement({
  actions: ['lambda:InvokeFunction'],
  resources: props.timeoutHandler.resourceArnsForGrantInvoke,
});
```

Is it because we control both the role and the lambdas so we can be certain they are in the same account?

Reply:
we can only call `grantInvoke` on something that implements `IGrantable` since we have the `role` available here, we can use `grantInvoke`. (In the step-functions-tasks cases this doesn't work because we don't have access to an `IGrantable` when the policy statements are being created.  Does that answer the question?